packages/wrangler/src/config/validation-helpers.ts

```diff
@@ -589,6 +589,16 @@ const isRecord = (
 ): value is Record<string | number | symbol, unknown> =>
 	typeof value === "object" && value !== null && !Array.isArray(value);
 
+export const warnForUnnecessaryProperies = (diagnostic: Diagnostics, field: string, value: Object, keys: string[]) => {
+	for (const prop of Object.keys(value)) {
+		if(!keys.includes(prop)) {
+			diagnostic.warnings.push(
+				`"${field}" has unexpected key '${prop}'. Allowed keys are${keys.map(key => ` '${key}'`)}.`
+			)
+		}
+	}
+}
+
```

Comment:
Did you see this helper?

https://github.com/cloudflare/workers-sdk/blob/9aa155d43a3ad36ab620efa236a9ba2357bbde5a/packages/wrangler/src/config/validation-helpers.ts#L516-L534

Is it possible to use that instead?

Reply:
Didn't spot it! Thanks for pointing it out :)